msal/src/Microsoft.Identity.Client/ClaimsPrincipalExtension.cs

```diff
@@ -0,0 +1,108 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Security.Claims;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Extension methods for ClaimsPrincipal
+    /// </summary>
+    public static class ClaimsPrincipalExtension
```

Comment:
>ClaimsPrincipalExtension [](start = 24, length = 24)

what about .net desktop?

Reply:
I don't understand your question - this class is now compiled on all our target frameworks, including .net classic. Should it be added only to .net core / .net standard?

Reply:
IMO, this only makes sense in web scenarios. perhaps I'm missing something.

---
In reply to: [237044980](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1435#discussion_r237044980) [](ancestors = 237044980)

Reply:
indeed it's only for Web scenarios.